candlepin/candlepin
# Adjust pom generation to handle spec-first changes

Adds a pom task to the api sub-project, which does the following:

- uses openapi generator maven plugin to generate code
- uses dependencyManagement gradle plugin to include dependencyManagement in the pom.

Applies the spec version plugin to the API sub-project, so that the API version ends up with a proper version.

Updates generated pom.xml files.

To test:

```
./gradlew pom; mvn package
```

Without this change, the above fails; with it, it succeeds.

NOTE: While reviewing #2612, I encountered issues trying to verify that Maven builds are equivalent, and I found the need for this PR.
